cluster/addons.go

```diff
@@ -196,11 +212,15 @@ func (c *Cluster) doAddonDeploy(ctx context.Context, addonYaml, resourceName str
 	addonJob, err := addons.GetAddonsExcuteJob(resourceName, node.Name, c.Services.KubeAPI.Image)
 
 	if err != nil {
-		return fmt.Errorf("Failed to deploy addon execute job: %v", err)
+		return fmt.Errorf("Failed to generate addon execute job: %v", err)
 	}
 	err = c.ApplySystemAddonExcuteJob(addonJob)
 	if err != nil {
-		return fmt.Errorf("Failed to deploy addon execute job: %v", err)
+		if isCritical {
+			return &addonError{fmt.Sprintf("Failed to deploy addon execute job [%s]: %v", resourceName, err), isCritical}
+		}
+		log.Warnf(ctx, "Failed to deploy addon execute job [%s]: %v", resourceName, err)
+		return &addonError{"", isCritical}
```

Comment:
I'd rather return the addonError in a generic manner, and let the caller stack make a decision on how to handle the error (log it in Warn vs propagating the error up). Try to avoid logging in the subcalls.